main_loop_tf/main.py

```diff
@@ -490,64 +494,59 @@ def build_graph(placeholders, input_shape, optimizer, weight_decay, loss_fn,
     devices = cfg.devices
     nclasses = cfg.nclasses
     global_step = cfg.global_step
+    is_training = which_set == 'train'
+    reuse_variables = not is_training
+
     if is_training:
         [inputs_per_gpu, labels_per_gpu, prev_err] = placeholders
+        summaries = tf.get_collection_ref(key='train_summaries')
     else:
         [inputs_per_gpu, labels_per_gpu] = placeholders
-
+        summaries = tf.get_collection_ref(
+            key='val_' + which_set + '_summaries')
```

Comment:
If which_set already contains the 'eval_' prefix probably we could avoid 'val_' for the collection name.